### PR TITLE
Create 6-5-custom-contraction_cn.ipynb

### PR DESCRIPTION
**Pull request by SexyCarrots**

### CODE REVIEW

File: docs/source/whitepaper/6-5-custom-contraction.ipynb [hunk 1]
```diff
@@ -83,9 +83,9 @@
     "    \n",
     "   * **SIZE**: the size of the largest intermediate tensor stored in memory.\n",
     "\n",
-    "Since simulations in TensorCircuit are AD-enabled, where all intermediate results need to be cached and traced, the more relevant spatial cost metric is writes instead of size.\n",
+    "Since simulations in TensorCircuit are AD-enabled, where all intermediate results need to be cached and traced, the more relevant spatial cost metric is written instead of size.\n",
```
Comment (refraction-ray) [inline docs/source/whitepaper/6-5-custom-contraction.ipynb:86]: revert to writes, "writes" is a metric name

File: docs/source/whitepaper/6-5-custom-contraction_cn.ipynb [hunk 2]
```diff
@@ -0,0 +1,281 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "id": "bc07c2b3",
+   "metadata": {},
+   "source": [
+    "# 定制收缩"
```
Comment (refraction-ray) [inline docs/source/whitepaper/6-5-custom-contraction_cn.ipynb:8]: 定制收缩路径

File: docs/source/whitepaper/6-5-custom-contraction_cn.ipynb [hunk 3]
```diff
@@ -0,0 +1,281 @@
...
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "918c848d",
+   "metadata": {},
+   "source": [
+    "## 设置\n",
+    "\n",
+    "cotengra请参考[安装文档](https://cotengra.readthedocs.io/en/latest/installation.html)，由于没有上传到PyPI，所以无法通过\n",
```
Comment (refraction-ray) [inline docs/source/whitepaper/6-5-custom-contraction_cn.ipynb:28]: cotengra 安装请参考

File: docs/source/whitepaper/6-5-custom-contraction_cn.ipynb [hunk 4]
```diff
@@ -0,0 +1,281 @@
...
+  },
+  {
+   "cell_type": "markdown",
+   "id": "17e9c8bc",
+   "metadata": {},
+   "source": [
+    "opt-einsum 提供了几个收缩优化器，并随 TensorNetwork 包一起提供。由于 TensorCircuit 建立在 TensorNetwork 之上，我们可以使用这些简单的收缩优化器。\n",
+    "尽管对于任何中等系统，只有贪婪优化器有效，但其他优化器具有指数缩放并且在电路模拟场景中失败。\n",
+    "\n",
+    "在本说明中，我们始终为收缩系统设置 ``contraction_info=True``（默认为 ``False``），它将打印收缩信息摘要，包括收缩大小、触发器和写入。\n",
```
Comment (refraction-ray) [inline docs/source/whitepaper/6-5-custom-contraction_cn.ipynb:80]: 使用英文：size flops 和 writes

File: docs/source/whitepaper/6-5-custom-contraction_cn.ipynb [hunk 5]
```diff
@@ -0,0 +1,281 @@
...
+   "source": [
+    "opt = ctg.ReusableHyperOptimizer(\n",
+    "    methods=[\"greedy\", \"kahypar\"],\n",
+    "    parallel=True,\n",
+    "    minimize=\"write\",\n",
+    "    max_time=120,\n",
+    "    max_repeats=1024,\n",
+    "    progbar=True,\n",
+    ")\n",
+    "# 注意：目前，parallel 仅适用于较新版本的 python 中的 “ray”\n",
```
Comment (refraction-ray) [inline docs/source/whitepaper/6-5-custom-contraction_cn.ipynb:177]: 对于新版本 python，仅 "ray" 选项对于 parallel 参数适用

File: docs/source/whitepaper/6-5-custom-contraction_cn.ipynb [hunk 6]
```diff
@@ -0,0 +1,281 @@
...
+    "tc.set_contractor(\n",
+    "    \"custom\", optimizer=opt, preprocessing=True, contraction_info=True, debug_level=2\n",
+    ")\n",
+    "# opt-einsum 兼容函数接口作为优化器的参数传递\\\n",
+    "# 还要注意 preprocessing=True 如何将单个量子比特门合并到相邻的两个量子比特门中\n",
```
Comment (refraction-ray) [inline docs/source/whitepaper/6-5-custom-contraction_cn.ipynb:182]: 如何->可以

File: docs/source/whitepaper/6-5-custom-contraction_cn.ipynb [hunk 7]
```diff
@@ -0,0 +1,281 @@
...
+    "testbed()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "871115c1",
+   "metadata": {},
+   "source": [
+    "我们甚至可以在路径搜索之后包含收缩重新配置，这进一步大大提高了收缩路径的空间效率。"
```
Comment (refraction-ray) [inline docs/source/whitepaper/6-5-custom-contraction_cn.ipynb:191]: 收缩重新配置 -> reconfigure

File: docs/source/whitepaper/6-5-custom-contraction_cn.ipynb [hunk 8]
```diff
@@ -0,0 +1,281 @@
...
+    "    tree = opt.search(inputs, output, size)\n",
+    "    tree_r = tree.subtree_reconfigure_forest(\n",
+    "        progbar=True, num_trees=10, num_restarts=20, subtree_weight_what=(\"size\",)\n",
+    "    )\n",
+    "    return tree_r.get_path()\n",
+    "\n",
+    "\n",
+    "# subtree_reconfigure_forest 还有一个默认的 parallel=True 选项，\n",
+    "# 对于上面的较新版本的 python，这只能设置为 “ray”\n",
+    "# 请注意不同版本的 cotengra 在最后一行中如何破坏 API：get_path 或 pat\n",
```
Comment (refraction-ray) [inline docs/source/whitepaper/6-5-custom-contraction_cn.ipynb:246]: 中 API 如何发生了改变

File: docs/source/whitepaper/6-5-custom-contraction_cn.ipynb [hunk 9]
```diff
@@ -243,7 +243,7 @@
     "\n",
     "# subtree_reconfigure_forest 还有一个默认的 parallel=True 选项，\n",
     "# 对于上面的较新版本的 python，这只能设置为 “ray”\n",
-    "# 请注意不同版本的 cotengra 在最后一行中如何破坏 API：get_path 或 pat\n",
+    "# 请注意不同版本的 cotengra 在最后一行中 API 如何发生了改变：get_path 或 pat\n",
```
Comment (refraction-ray) [inline docs/source/whitepaper/6-5-custom-contraction_cn.ipynb:246]: path

Comment (refraction-ray) [reply docs/source/whitepaper/6-5-custom-contraction_cn.ipynb:246]: and also remember fix the typo "pat" to "path" in the English ipynb

**Comment by SexyCarrots** [comment]
# Updated

**Review by refraction-ray — APPROVED** [review]
LGTM

**Pull request merged**